library/std/src/sys/unix/fd.rs

```diff
@@ -28,7 +33,9 @@ const READ_LIMIT: usize = libc::ssize_t::MAX as usize;
 
 impl FileDesc {
     pub fn new(fd: c_int) -> FileDesc {
-        FileDesc { fd }
+        assert_ne!(fd, -1i32);
```

Comment:
Will there be `unsafe fn new_unchecked`, to avoid including panicking code in a program that otherwise avoid it?